rotkehlchen/history/events/structures/eth2.py

```diff
@@ -207,25 +216,63 @@ def process(
-            notes=f'{name} of {self.amount} ETH from validator {self.validator_index}. Only {profit_amount} is profit',  # noqa: E501
-            location=self.location,
-            timestamp=self.get_timestamp_in_sec(),
-            asset=self.asset,
-            amount=profit_amount,
-            taxable=accounting.events_accountant.rules_manager.eth_staking_taxable_after_withdrawal_enabled,
-        )
+        with accounting.database.conn.read_ctx() as cursor:
+            validator_info = accounting.dbeth2.get_validators_with_status(
+                cursor=cursor,
+                validator_indices={self.validator_index},
+            )[0]
+
+        is_exit = self.is_exit_or_blocknumber != 0
+        name = 'Exit' if is_exit else 'Withdrawal'
+        event_ts = self.get_timestamp_in_sec()
+        if validator_info.validator_type != ValidatorType.ACCUMULATING:
+            # Non-accumulating validator: Compare exit amount to MIN_EFFECTIVE_BALANCE
+            profit_or_loss_amount = self.amount
+            if self.amount >= MIN_EFFECTIVE_BALANCE:
+                profit_or_loss_amount = self.amount - MIN_EFFECTIVE_BALANCE
+        elif is_exit:  # Accumulating validator exit
+            if (profit_or_loss_amount := max(ZERO, self.amount - MAX_EFFECTIVE_BALANCE)) == ZERO:
+                # If no profit detected, then it is a loss and equals total deposited - total withdrawn(just before this event)  # noqa: E501
+                balances, _, _ = accounting.dbeth2.process_accumulating_validators_balances_and_pnl(  # noqa: E501
+                    from_ts=Timestamp(0),
+                    to_ts=Timestamp(event_ts - 1),
+                    validator_indices=[self.validator_index],
+                    balances_over_time=defaultdict(lambda: defaultdict(lambda: ZERO)),
+                    withdrawals_pnl=defaultdict(lambda: ZERO),
+                    exits_pnl=defaultdict(lambda: ZERO),
+                )
+                profit_or_loss_amount = -FVal(list(balances[self.validator_index].values())[-1] - self.amount)  # noqa: E501
+        else:  # Accumulating validator withdrawal (partial or skimming)
+            _, withdrawal_pnl, _ = accounting.dbeth2.process_accumulating_validators_balances_and_pnl(  # noqa: E501
+                from_ts=event_ts,
+                to_ts=Timestamp(event_ts + 1),
+                validator_indices=[self.validator_index],
+                balances_over_time=defaultdict(lambda: defaultdict(lambda: ZERO)),
+                withdrawals_pnl=defaultdict(lambda: ZERO),
+                exits_pnl=defaultdict(lambda: ZERO),
+            )
+            profit_or_loss_amount = withdrawal_pnl.get(self.validator_index, ZERO)
+
+        if profit_or_loss_amount > ZERO:
+            accounting.add_in_event(
+                event_type=AccountingEventType.HISTORY_EVENT,
+                notes=f'{name} of {self.amount} ETH from validator {self.validator_index}. Only {profit_or_loss_amount} is profit',  # noqa: E501
```

Comment:
The `Only {profit_or_loss_amount} is profit` should only be added there if amount and pnl are not the same. Which will only happen for exits and for accumulating withdrawals(?)

Reply:
updated it.